code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/filter/settings/SettingsFilter.java

```diff
@@ -4,21 +4,26 @@
 import io.cattle.platform.api.utils.ApiUtils;
 import io.cattle.platform.core.addon.ActiveSetting;
 import io.cattle.platform.core.model.Setting;
+import io.github.ibuildthecloud.gdapi.context.ApiContext;
 import io.github.ibuildthecloud.gdapi.request.ApiRequest;
 import io.github.ibuildthecloud.gdapi.util.RequestUtils;
 
+import java.util.Collections;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
 import org.apache.commons.collections.Predicate;
 
+import com.netflix.config.DynamicStringMapProperty;
+
 public class SettingsFilter implements Predicate {
 
     Set<String> publicSettings;
     boolean all;
     boolean canListAll;
+    public static final DynamicStringMapProperty SETTINGS_HIDDEN = new DynamicStringMapProperty("settings.hidden", Collections.emptyMap());
```

Comment:
We don't want a setting as then the setting could just be overidden to unhide. Just a hard coded list in the code